client/landing/gutenboarding/onboarding-block/import/index.tsx

```diff
@@ -0,0 +1,105 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+import { useHistory } from 'react-router-dom';
+import { useI18n } from '@automattic/react-i18n';
+import { Title } from '@automattic/onboarding';
+import { Button, Modal, TextControl } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+import { validateImportUrl } from 'calypso/lib/importer/url-validation';
+import url from 'url'; // eslint-disable-line no-restricted-imports
+
+/**
+ * Internal dependencies
+ */
+import { STORE_KEY } from '../../stores/onboard';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const ImportStep: React.FunctionComponent = () => {
+	const { __ } = useI18n();
+
+	const history = useHistory();
+	const { resetOnboardStore, setImportUrl } = useDispatch( STORE_KEY );
+	const { importUrl } = useSelect( ( select ) => select( STORE_KEY ).getState() );
+
+	const [ validationError, setValidationError ] = useState( '' );
+
+	const goBack = () => {
+		resetOnboardStore();
```

Comment:
Please note that `resetOnboardStore` action will reset the entire `Onboard` data-store, including persisted selections (site title, design, font, domain, plan). Since Gutenboarding is a non-linear flow, users are able to do all selections and then for example: navigate all the way back to `AcquireIntent` step, open importer modal, close it and continue to site creation.

Reply:
Thanks for noting this, @razvanpapadopol! I'll make it a little more nuanced. 😉